packages/@lwc/engine-core/src/framework/api.ts

```diff
@@ -377,12 +382,17 @@ export function s(
     ) {
         children = slotset[slotName];
     }
-    const vnode = h('slot', data, children);
-    if (vnode.owner.renderer.syntheticShadow) {
+    const vmBeingRendered = getVMBeingRendered()!;
+
+    if (!hasShadow(vmBeingRendered)) {
```

Comment:
I still think `hasShadow` is the wrong name for this function, something like: `isLightRenderMode(vmBeingRendered)` is a lot more flexible for the future.